src/amberclient/common/amber_proxy.py

```diff
@@ -69,4 +79,55 @@ def terminate_proxy(self):
         driver_msg_builder = drivermsg_pb2.DriverMsg()
         driver_msg_builder.type = drivermsg_pb2.DriverMsg.CLIENT_DIED
 
-        self.__amber_client.send_message(self.build_header(), driver_msg_builder)
\ No newline at end of file
+        self.__amber_client.send_message(self.build_header(), driver_msg_builder)
+
+    def get_next_syn_num(self):
+        self.__syn_num_lock.acquire()
+        try:
+            self.__syn_num += 1
+            return self.__syn_num
+        finally:
+            self.__syn_num_lock.release()
+
+    def get_future_object(self, ack_num):
+        obj = None
+        self.__future_objs_lock.acquire()
+        try:
+            if ack_num in self.__future_objs:
+                obj = self.__future_objs[ack_num]
+                del self.__future_objs[ack_num]
+        finally:
+            self.__future_objs_lock.release()
+        return obj
```

Comment:
Usage of this method:

```
obj = self.get_future_object(message.ackNum)
if isinstance(obj, Status):
    DummyProxy.__fill_status(obj, message)
```

If `obj` is `None` then `isinstance()` return `False`.